### DISCUSSION TITLE
How do I disable specific default keymap without overriding it?

### DISCUSSION THREAD

**Discussion by Demianeen** [LazyVim/LazyVim#2912] [Q&A] [answered]
I installed mini.move plugin and I wanted to set moving line left and right on H and L. I pass the plugin keymaps like this:

```lua
return {
  'echasnovski/mini.move',
  event = 'VeryLazy',
  keys = {
	-- trying to disable H and L keymaps here
    { '<s-h>', false },
    { '<s-l>', false },
  },
  opts = {
    mappings = {
      -- Move visual selection in Visual mode. Defaults are Alt (Meta) + hjkl.
      left = 'H',
      right = 'L',
      down = 'J',
      up = 'K',

      -- Move current line in Normal mode
      line_left = 'H',
      line_right = 'L',
      -- line_down = 'J',
      -- line_up = 'K',
    },
  },
}
```

But it doesn't help, I still have next, prev buffer keymaps set. How do you think I can disable these two keymaps?

**Comment by dpetka2001** [comment]
You can only use `keys` section to disable keymaps that have been defined in the specific spec of a plugin. You can't use it to disable keymaps in general.

The process I usually follow to disable keymaps I don't like, is search through the repo to see where they are defined. For example if you search for `<s-h>`, you get [these results](https://github.com/search?q=repo%3ALazyVim%2FLazyVim%20%3Cs-h%3E&type=code). From there you see that the specific keymap is defined in 2 places (once in `keymaps.lua` for the default `bprevious` command and one more in `ui.lua` in the Bufferline spec). 

So, to disable this keymap as you want, you would have to do it in 2 places as well. First in your own `keymaps.lua` and second in your custom spec for Bufferline in `/lua/plugins/` directory.

**Comment by andreacfromtheapp** [comment]
here's how I do it to map lateral buffer movement to ALT-arrows, I hope it helps:

```lua
-- lua.config.keymaps
vim.keymap.del("n", "<S-h>")
vim.keymap.del("n", "<S-l>")
vim.keymap.set("n", "<M-Left>", "<cmd>bprevious<cr>", { desc = "Prev buffer" })
vim.keymap.set("n", "<M-Right>", "<cmd>bnext<cr>", { desc = "Next buffer" })
```

**Reply by Demianeen** [reply, marked as answer]
It does remove the keymap! Mini.move plugin is fired before lazyvim, so it never applies the keymap through.

In my keys only manually overriding in keys did help:
```lua
  keys = {
    {
      'H',
      function()
        require('mini.move').move_line('left')
      end,
    },
    {
      'L',
      function()
        require('mini.move').move_line('right')
      end,
    },
  },
```

**Reply by dpetka2001** [reply]
@Demianeen 
You could also just defer the setup of `mini.move` like so 
```lua
  config = function(_, opts)
    vim.schedule(function()
      require("mini.move").setup(opts)
    end)
  end,
```
This would make it so, that the plugin gets loaded after the `VeryLazy` event (where `keymaps.lua` also gets loaded), thus your changes in `opts.mappings` would take effect.

**Reply by Demianeen** [reply]
Thanks @dpetka2001!
